Update cloud platform packages.yaml files

ulmononian opened the pull request:
This PR contains updates to the packages.yaml files for the NOAA cloud platforms (aws, azure, and gcp). The changes are relatively minimal, with the most significant adjustment being the switch for some external library prefixes to EPIC's new miniconda3 installation path. Other changes include pointers to updated git/git-lfs installation, as well as modified curl external configurations. These updates were catalyzed by various failures and errors (specifically with building cmake 3.22.1) caused when using the libraries provided by the current miniconda/python pointed to in these YAMLs, as well as repeated failures resulting from the current git and curl externals.

Installations using the updated package YAMLs in this PR were tested on aws, azure, and gcp w/ the ufs-srw-dev template (including the met/metplus additions from #374).

ulmononian committed:
Add ufs-wm-dev template to develop.

ulmononian committed:
Remove ufs-wm-dev template to get fork dev aligned with main dev.

ulmononian committed:
Merge branch 'NOAA-EMC:develop' into develop

Commit:
Add updated noaa-gcloud packages.yaml

ulmononian committed:
Update packages.yaml

Update version of openssl provided by miniconda.

ulmononian committed:
Update packages.yaml

add curl note

ulmononian committed:
Update packages.yaml

add git note.

ulmononian committed:
Update packages.yaml

fix typo.

Commit:
Update packages.yaml for AWS.

Commit:
Merge branch 'feature/update_noaacloud' of https://github.com/ulmononian/spack-stack into feature/update_noaacloud

ulmononian committed:
Update packages.yaml

Remove miniconda module from python external due to issues it causes when setting up the stack-python meta-module.

ulmononian committed:
Update packages.yaml

Remove miniconda module from external python for same reason as with gcp packages.yaml.

ulmononian committed:
Update packages.yaml

set buildable to false for git; add curl as an external

ulmononian committed:
Update packages.yaml

add note for curl

ulmononian committed:
Update packages.yaml

fix buildable flag indentation

ulmononian committed:
Update packages.yaml

ulmononian committed:
Update packages.yaml

update azure packages.yaml

ulmononian committed:
Update packages.yaml

remove bin from curl and git paths

climbfuji commented on configs/sites/noaa-azure/packages.yaml:
```diff
@@ -45,7 +45,6 @@ packages:
   berkeley-db:
     externals:
     - spec: berkeley-db@5.3.21
-      prefix: /usr
```
How is this working? I thought that spack either needs a prefix or a module to load for external packages?

ulmononian replied:
@climbfuji whoops...this was inadvertent. i did this commit in a roundabout way due to github issues on azure. fixed now.

climbfuji reviewed (commented):
I am having a bit of a problem with the directory structure - why do we need to replace the institution-independent `spack-stack` directory with `EPIC`? `JCSDA also wants to use the NOAA cloud environment in the next year, and we would definitely prefer an institution-independent structure. Unless we agree to have separate/duplicate site configs, which doesn't seem ideal. Let's discuss this before we make changes or merge as-is.

ulmononian committed:
Update packages.yaml

fix berkeley-db prefix omission

ulmononian committed:
Update packages.yaml

ulmononian commented:
@climbfuji regarding the institutional independence: that's a very valid question and point. we ended up making these changes internally because the current package YAML files for the cloud sites were leading to several points of failure for srw and wm stack installs, one of which was the reference to the miniconda & python in the current develop versions of the files. we were unsure who installed/maintained the miniconda at /contrib/spack-stack/apps/miniconda, so we went ahead with our install. 

i'm happy to revert this to /contrib/spack-stack/apps/miniconda if the maintainer can either (i) let us know where we went wrong in trying to use packages.yaml as is to install ufs-srw-dev/ufs-srw-public-v2/ufs-wm-dev, or (ii)  update/fix the conda-provided packages to work with the spack envs. we need.

climbfuji commented:
Thanks, @ulmononian. I am not clear on how these containers are set up and used across NOAA and/or EPIC. It's probably best to have a quick chat early next week - I'll see if I can find an empty slot in our schedules.

ulmononian committed:
Merge branch 'NOAA-EMC:develop' into feature/update_noaacloud

ulmononian committed:
Update packages.yaml

update miniconda for python external

ulmononian committed:
Update packages.yaml

Update miniconda/python and associated externals to new /contrib/spack-stack miniconda installation.

ulmononian committed:
Update packages.yaml

Revert git-lfs external.

ulmononian committed:
Update packages.yaml

revert to /usr git for now.

ulmononian committed:
Update packages.yaml

Update to new miniconda/python. Remove curl. Revert to old git (for now).

ulmononian committed:
Update packages.yaml

Update to new miniconda python. Revert to old git (temporarily). Use existing git-lfs.

ulmononian commented:
@climbfuji i installed a new miniconda/python in /contrib/spack-stack and made the changes in the packages.yaml files for the cloud platforms. i also removed the paths to external git/curl libraries that were installed in the EPIC space. while the git that is currently pointed to is pretty old, spack has lately been able to clone properly w/ it or it builds a newer git. pointing to the curl in /usr/bin never seems to work, so right now, i just have spack build it. something that can be done is to install new git & curl in /contrib/spack-stack/apps/utils, but i think that can come after this PR if it is wanted.

ulmononian committed:
Merge branch 'NOAA-EMC:develop' into feature/update_noaacloud

ulmononian commented:
@climbfuji thanks for the review on this, dom. perhaps we can touch base sometime in the coming week or so about the git/curl issues.

ulmononian commented:
@climbfuji is there more to be done on this one before it can be merged? i was not sure about the ci tests...

climbfuji commented:
Sorry, I was waiting for approval from EMC, but let's merge this. I will update from develop quickly, if everything looks ok (no need to wait for CI for these files), then I will merge. Sorry for the delay.

climbfuji committed:
Merge branch 'develop' into feature/update_noaacloud

ulmononian commented:
> Sorry, I was waiting for approval from EMC, but let's merge this. I will update from develop quickly, if everything looks ok (no need to wait for CI for these files), then I will merge. Sorry for the delay.

no worries at all, @climbfuji!! i just wanted to make sure i was not forgetting to address something or missing a step in the standard spack-stack PR review process. thanks so much for reviewing and getting this merged.
